src/services/email/handlers/processEmails.tsx

```diff
@@ -0,0 +1,116 @@
+import { DateTime } from "luxon";
+import { decode } from "base-64";
+import {
+  CognitoIdentityProviderClient,
+  ListUsersCommand,
+  // UserType as CognitoUserType,
+} from "@aws-sdk/client-cognito-identity-provider";
+import { SESClient, SendTemplatedEmailCommand } from "@aws-sdk/client-ses";
+
+import { KafkaEvent } from "shared-types";
+
+// import { CognitoUserAttributes } from "shared-types";
+
+const SES = new SESClient({ region: process.env.region });
+const Cognito = new CognitoIdentityProviderClient({
+  region: process.env.region,
+});
+
+const emailsToSend = {
+  "initial-submission-medicaid-spa": [{
+    "templateBase": "initial-submission-medicaid-spa-cms",
+    "sendTo": [process.env.osgEmail],
+  }, {
+    "templateBase": "initial-submission-medicaid-spa-state",
+    "sendTo": ["submitterEmail"],
+  }],
+  "initial-submission-chip-spa": [{
+    "templateBase": "initial-submission-chip-spa-cms",
+    "sendTo": [process.env.chipEmail],
+    "addCC": [`${process.env.chipCCList}`],
+  }, {
+    "templateBase": "initial-submission-chip-spa-state",
+    "sendTo": ["submitterEmail"],
+  }],
+}
+
+const createSendTemplatedEmailCommand = (data) =>
+  new SendTemplatedEmailCommand({
+    Source: process.env.emailSource ?? "kgrue@fearless.tech",
+    Destination: {
+      ToAddresses: data.ToAddresses,
+      CcAddresses: data.CcAddresses,
+    },
+    TemplateData: JSON.stringify({ applicationEndpoint: "onemac.cms.gov", ...data }),
+    Template: `${data.templateBase}_${process.env.stage}`,
+    ConfigurationSetName: process.env.emailConfigSet,
+  });
+
+export const main = async (event: KafkaEvent) => {
+  console.log("Received event (stringified):", JSON.stringify(event, null, 4));
+
+  const emailQueue: any[] = [];
+  const getStateUsersFor: string[] = [];
+  Object.values(event.records).forEach((oneSource) =>
+    oneSource.forEach((encodedRecord) => {
+      if (!encodedRecord.value) return;
+      const record = { id: decode(encodedRecord.key), ...JSON.parse(decode(encodedRecord.value)) };
+      console.log("here is the decoded record: ", record);
+      if (record?.origin !== "micro") return;
+      if (!record?.actionType) record.actionType = "initial-submission";
```

Comment:
This follows the logic from the data sink at the time I was building this service.  I fully expect this to change, and I am hoping/predicting that it changes to where action-type is always sent and the default is not triggered.  

The check for "micro" limits the emails to only events marked as micro, and, so far, the only events I've seen without action types are the initial-submissions.

The initial emails to be built are initial-submission emails, I personally could not see any other way to identify them, though, I admit, I did not spend a lot of time thinking about this once I decided it would probably change on the event side soon enough....

Reply:
I put it on the record *because* that's where I feel like it should be coming from.  Like this is a bandaid and in reality, it will come from record, so should be on the record now?  That was my reasoning, but it can be changed if you really want.

I did update "initial-submission" to match "new-submission"